torch/csrc/jit/script/compiler.cpp

```diff
@@ -2774,6 +2773,23 @@ struct to_ir {
       Value* tuple_val,
       Value* idx_val) {
     auto tuple_typ = tuple_val->type()->cast<TupleType>();
+    auto elems = tuple_typ->elements();
+    if (!toIValue(idx_val) && elems.size() > 0) {
+      auto list = tryConvertToType(
```

Comment:
If someone iterates over a list in constants won't it re-create the list here every iteration? Maybe we could fix this closer to the source of the issue and allow `List` constants by converting lists to some `ConstantList` subclass of `list` where the`__setitem__` throws an error

Reply:
I'm not sure we want to add the overhead of a Constant List subclass, that would be difficult. How would you handle ConstantLists passed into a function that takes in a list and enforce that it does not get mutated ? 

This is an example of where Loop Invariant Code Motion would be useful.